### PR TITLE
feat: autocreate project with tags

### PR DESCRIPTION
**Pull request by JCHacking**
### Description
To be able to upload a bom and auto-create the project with specific tags

### Addressed Issue
https://github.com/DependencyTrack/dependency-track/issues/1674

### Additional Details
I have added the optional property to the PUT method to upload BOM files, I have not programmed much in Java so any suggestion to improve this PR is welcome.
Thanks!!!

### Checklist
- [X] I have read and understand the [contributing guidelines](../CONTRIBUTING.md#pull-requests)
- [ ] This PR fixes a defect, and I have provided tests to verify that the fix is effective
- [X] This PR implements an enhancement, and I have provided tests to verify that it works as intended
- [ ] This PR introduces changes to the database model, and I have added corresponding [update logic](https://github.com/DependencyTrack/dependency-track/tree/master/src/main/java/org/dependencytrack/upgrade)
- [ ] This PR introduces new or alters existing behavior, and I have updated the [documentation](https://github.com/DependencyTrack/dependency-track/tree/master/docs/_docs) accordingly

### CODE REVIEW

**Comment by valentijnscholten** [comment]
I like it. Can you add a test with tags populated?

**Comment by JCHacking** [comment]
Sure! Tests added

File: src/main/java/org/dependencytrack/resources/v1/vo/BomSubmitRequest.java [hunk 1]
```diff
@@ -51,6 +53,8 @@ public final class BomSubmitRequest {
     @Pattern(regexp = RegexSequence.Definition.PRINTABLE_CHARS, message = "The project version may only contain printable characters")
     private final String projectVersion;
 
+    private final List<Tag> projectTags;
```
Comment (nscuro) [inline src/main/java/org/dependencytrack/resources/v1/vo/BomSubmitRequest.java:56]: We should probably accept a list of `String` here instead. From the client's perspective, passing this:

```json
{
  "projectTags": ["foo", "bar"]
}
```

is more straightforward than:

```json
{
  "projectTags": [
    {
      "name": "foo"
    },
    {
      "name": "bar"
    }
  ]
}
```

File: src/main/java/org/dependencytrack/resources/v1/vo/BomSubmitRequest.java [hunk 2]
```diff
@@ -110,6 +117,10 @@ public String getProjectVersion() {
         return projectVersion;
     }
 
+    public List<Tag> getProjectTags() {
```
Comment (nscuro) [inline src/main/java/org/dependencytrack/resources/v1/vo/BomSubmitRequest.java:120]: Can you add a `@Schema(description = "")` here, where `description` explains that the tags defined here will **overwrite** any tags on existing projects that were manually applied?

It was a discussion point in #1674 so I believe the API docs should call out the behavior.

**Review by nscuro — CHANGES_REQUESTED** [review]

File: src/main/java/org/dependencytrack/resources/v1/vo/BomSubmitRequest.java (hunk 1 above)
Comment (JCHacking) [reply src/main/java/org/dependencytrack/resources/v1/vo/BomSubmitRequest.java:56]: I agree, but I also think that using Tag keeps the API consistent, as in other methods this happens.

For example:
```json
PUT /v1/project

{
  "tags": [
    {
      "name": "LNsN}MPtIehaapsLeihSsMtMctLN{_aNcipcseIaLSWaNs}t_IesSSSsicpe_hIWtL_McIhs{L}P_SPc}S_M}{S}peWhcP_PeWhs"
    }
  ],
  ...
}
```


Do I make the change to make it easier or do I leave it like this to make the api more consistent?

**Comment by JCHacking** [comment]
I see that the test has failed, I imagine that it is because it is not the same object since one is the tags only with name and the others are the tags obtained from the persistence that will have more properties.

I should only make an assertEquals of the list of strings with the name of the tags, right?

File: src/main/java/org/dependencytrack/resources/v1/vo/BomSubmitRequest.java (hunk 1 above)
Comment (nscuro) [reply src/main/java/org/dependencytrack/resources/v1/vo/BomSubmitRequest.java:56]: Good point. Let's leave it consistent then.

**Comment by nscuro** [comment]
> I should only make an assertEquals of the list of strings with the name of the tags, right?

Yes. Consider using AssertJ's fluent assertions, like so:

```java
assertThat(project.getTags())
    .extracting(Tag::getName)
    .containsExactlyInAnyOrder("foo", "bar");
```

**Comment by JCHacking** [comment]
Fixed it with your suggestion, thanks!!

**Review by nscuro — APPROVED** [review]
Thanks!

**Pull request merged**